writing_your_start-up_command.md

```diff
@@ -80,20 +80,28 @@ Explain. Also explain you can assign an alias to the script itself. FIXME
 
 Each directory within music will show up as a crate inside xwax, etc. FIXME
 
+(Fixed so it can take folders names with spaces and some illegal characters.)
+
 Should this be it's own page?
 
 ```bash
-#!/bin/sh
+#!/bin/bash
 
 # The loop
 # Replace ~/Music/* with the location of the directories you wish to scan FIXME
+
 PLAYLISTS=
-for P in ~/Music/*; do
-    PLAYLISTS="-l $P $PLAYLISTS"
+  for P in ~/Music/*; do
+    if [ -d "$P" ]; then
+     PLAYLISTS="-l \"$P\" $PLAYLISTS"
+    fi
 done
 
-# Replace this start up command with your own, except keeping $PLAYLISTS in place of the -l flag
-xwax -a plughw:Audio4DJ,0,0 -a plughw:Audio4DJ,0,1 $PLAYLISTS
+# Replace this start up command with your own, except keeping $(echo $PLAYLISTS) in place of the -l flag 
+# and changing the terminal to be used, for example lxterminal instead of gnome-terminal.
+
+gnome-terminal -e "xwax -a plughw:Audio4DJ,0,0 -a plughw:Audio4DJ,0,1 $(echo $PLAYLISTS)"
```

Comment:
for some reason it doesnt take the double quotes in the path with a starightforward xwax command through the script ( thus not taking spaces in the foldername), but it does if manually add the double quotes to the path without the script or through a new terminal in the script.

its some odd behaviour i dont understand.

I opened a thread about it in the mailing list a week or two ago.  ill link you the message.

https://sourceforge.net/p/xwax/mailman/message/35504960/

just go ahead and try the script with a straightforward xwax command and then calling xwax through a new terminal.